File: 1037-minimum-number-of-k-consecutive-bit-flips/1037-minimum-number-of-k-consecutive-bit-flips.py
class Solution(object):
    def minKBitFlips(self, nums, k):
        n, curr_window_flips, ans = len(nums), 0, 0
        for i in range(n):
            if i - k >= 0 and nums[i-k] == 2:
                curr_window_flips -= 1
            
            if (nums[i] + curr_window_flips) % 2 == 0:
                if i + k > n:
                    return -1
                ans += 1
                curr_window_flips += 1
                nums[i] = 2
        
        return ans

        

File: 1037-minimum-number-of-k-consecutive-bit-flips/1037-minimum-number-of-k-consecutive-bit-flips.java
class Solution {
    public int minKBitFlips(int[] nums, int k) {
        int n = nums.length, ans = 0;
        int currWindowFlips = 0;

        for(int i = 0; i<n; i++) {
            if(i - k >= 0 && nums[i - k] == 2) currWindowFlips--;

            if((nums[i] + currWindowFlips) % 2 == 0) {
                if(i+k > n) return -1;
                ans++;
                currWindowFlips++;
                nums[i] = 2;
            }
        }

        return ans;
    }
}